server/handler/query.go

```diff
@@ -0,0 +1,88 @@
+package handler
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/src-d/gitbase-playground/server/serializer"
+)
+
+type queryRequest struct {
+	Query string `json:"query"`
+	Limit int    `json:"limit"`
+}
+
+// genericVals returns a slice of interface{}, each one a pointer to a NullString
+func genericVals(nColumns int) []interface{} {
+	columnVals := make([]sql.NullString, nColumns)
+	columnValsPtr := make([]interface{}, nColumns)
+
+	for i := range columnVals {
+		columnValsPtr[i] = &columnVals[i]
+	}
+
+	return columnValsPtr
+}
+
+// Query returns a function that forwards an SQL query to gitbase and returns
+// the rows as JSON
+func Query(db *sql.DB) RequestProcessFunc {
+	return func(r *http.Request) (*serializer.Response, error) {
+		var queryRequest queryRequest
+		body, err := ioutil.ReadAll(r.Body)
+		defer r.Body.Close()
```

Comment:
https://golang.org/pkg/net/http/#Request
> The Server will close the request body. The ServeHTTP Handler does not need to.

Reply:
Fixed, thanks! I actually took it from https://github.com/src-d/code-annotation/blob/master/server/handler/assignments.go#L82